rest/doc_api.go

```diff
@@ -408,7 +408,7 @@ func (h *handler) handlePutDocReplicator2(docid string, roundTrip bool) (err err
 
 	doc, rev, err := h.db.PutExistingRev(newDoc, history, true)
 
-	if newDoc != nil && roundTrip {
+	if doc != nil && roundTrip {
```

Comment:
This change is correct, but it looks like we should also be checking if err != nil prior to doing roundTrip handling, similar to what's done in handlePutDoc.

Reply:
Sure Adam; Error check has been handled in the latest version.